File: cpp/Gestor_Vehiculos/Automovil.cpp
#include "Automovil.h"
#include <iostream>

// Primero mando los datos comunes al constructor de Vehiculo (para que
// el "padre" se termine de armar), y despues guardo lo que es solo de
// Automovil (puertas, electrico).
Automovil::Automovil(const std::string& placa, const std::string& marca, const std::string& modelo,
                       int anio, double precio, bool disponible, int numeroPuertas, bool electrico)
    : Vehiculo(placa, marca, modelo, anio, precio, disponible),
      numeroPuertas(numeroPuertas), electrico(electrico) {
}

// Esta es la version de mostrarInformacion() que le toca a Automovil.
// Uso mostrarDatosComunes() para no repetir lo que ya imprime cualquier
// vehiculo, y despues agrego lo que es solo de un auto.
void Automovil::mostrarInformacion() const {
    std::cout << "----------------------------------------" << std::endl;
    std::cout << "Tipo: Automovil" << std::endl;
    mostrarDatosComunes();
    std::cout << "Numero de Puertas: " << numeroPuertas << std::endl;
    std::cout << "Electrico: " << (electrico ? "Si" : "No") << std::endl;
    std::cout << "----------------------------------------" << std::endl;
}

// Getters y setters de los datos propios de Automovil.
int Automovil::getNumeroPuertas() const { return numeroPuertas; }
void Automovil::setNumeroPuertas(int numeroPuertas) { this->numeroPuertas = numeroPuertas; }
bool Automovil::isElectrico() const { return electrico; }
void Automovil::setElectrico(bool electrico) { this->electrico = electrico; }

File: java/Gestor_Vehiculos/Automovil.java
package Gestor_Vehiculos;


public class Automovil extends Vehiculo {
    private int numeroPuertas;
    private boolean electrico;

    public Automovil(String placa, String marca, String modelo, int anio, double precio, 
                     boolean disponible, int numeroPuertas, boolean electrico) {
        
        super(placa, marca, modelo, anio, precio, disponible);
        this.numeroPuertas = numeroPuertas;
        this.electrico = electrico;
    }

    
    @Override
    public void mostrarInformacion() {
        System.out.println("----------------------------------------");
        System.out.println("Tipo: Automovil");
        mostrarDatosComunes();
        System.out.println("Numero de Puertas: " + numeroPuertas);
        System.out.println("Electrico: " + (electrico ? "Si" : "No"));
        System.out.println("----------------------------------------");
    }

    public int getNumeroPuertas() { return numeroPuertas; }
    public void setNumeroPuertas(int numeroPuertas) { this.numeroPuertas = numeroPuertas; }
    public boolean isElectrico() { return electrico; }
    public void setElectrico(boolean electrico) { this.electrico = electrico; }
}